src/App.js

```diff
@@ -1,13 +1,51 @@
 import React, { Component } from 'react'
-import { Instructions } from './components'
+import { Instructions, Counter } from './components'
 import './styles/app.css'
 
 class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { userInput: '', initialCount: 0 }
+
+    this.handleChange = this.handleChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  handleChange(event) {
+    this.setState({ userInput: event.target.value })
+  }
+
+  handleSubmit() {
+    if (isNaN(this.state.userInput)) {
+      alert("This wasn't a number, try again")
+    } else {
+      this.setState(state => {
+        return { initialCount: +state.userInput }
```

Comment:
I like the use of `+` here. I normally use `Number()` and didn't realise you could just do this. 🥇